core/src/contact_info.rs

```diff
@@ -10,6 +10,9 @@ use std::net::{IpAddr, SocketAddr};
 /// Structure representing a node on the network
 #[derive(Serialize, Deserialize, Clone, Debug)]
 pub struct ContactInfo {
+    /// validator identity
+    pub validator_id: Pubkey,
```

Comment:
It'd be nice to support `Option<Pubkey>` here.  If `None` then use `id`.   This avoids shuttling around an extra 32 bytes  if a validator is not in a group